src/core_plugins/kibana/public/context/query.js

```diff
@@ -0,0 +1,154 @@
+import _ from 'lodash';
+
+import { fetchAnchor } from './api/anchor';
+import { fetchPredecessors, fetchSuccessors } from './api/context';
+import { QueryParameterActionsProvider } from './query_parameters';
+
+
+function QueryActionsProvider(es, Notifier, Private, Promise) {
+  const {
+    increasePredecessorCount,
+    increaseSuccessorCount,
+    setPredecessorCount,
+    setQueryParameters,
+    setSuccessorCount,
+  } = Private(QueryParameterActionsProvider);
```

Comment:
The intention was to treat two state constituents `queryParameters` and `rows` (which are populated with the query results) as separate concerns. To me that warrants separate files.